client/me/security-ssh-key/add-ssh-key-form.tsx

```diff
@@ -0,0 +1,43 @@
+import { Button } from '@automattic/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { ChangeEvent, useState } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextarea from 'calypso/components/forms/form-textarea';
+
+const PUBLIC_SSH_KEY_INPUT_ID = 'public_ssh_key';
+
+interface AddSSHKeyFormProps {
+	addSSHKey( args: { name: string; key: string } ): void;
+}
+
+export const AddSSHKeyForm = ( { addSSHKey }: AddSSHKeyFormProps ) => {
+	const [ publicSSHKey, setPublicSSHKey ] = useState( '' );
+
+	const { __ } = useI18n();
+
+	return (
+		<form
+			onSubmit={ ( event ) => {
+				event.preventDefault();
+				addSSHKey( { name: 'default', key: publicSSHKey } );
+			} }
+		>
+			<FormFieldset>
+				<FormLabel htmlFor={ PUBLIC_SSH_KEY_INPUT_ID }>{ __( 'Public SSH key' ) }</FormLabel>
+				<FormTextarea
+					required
+					id={ PUBLIC_SSH_KEY_INPUT_ID }
+					value={ publicSSHKey }
+					placeholder={ __( 'Paste the public key here' ) }
```

Comment:
Let's include the supported format(s) here and maybe even an example:

> Paste the public SSH key in format X here. It should start ssh-rsa ...

Maybe we can also do some lightweight client-side validation with a regex?